daemon/api_test.go

```diff
@@ -134,8 +130,7 @@ oem: {store: {id: %q}}
 }
 
 func (s *apiSuite) TestPackageInfoOneIntegration(c *check.C) {
-	d := New()
-	d.addRoutes()
+	newTestDaemon()
```

Comment:
in that sense, it's making things a little bit better because now it's more
obvious that there is more going on.

The "more going on" is, of course, the setting up of the routes for each of
the commands, and the test operates on the commands directly.

On Thu, Oct 29, 2015 at 4:30 PM, Gustavo Niemeyer <notifications@github.com>
wrote:

> In daemon/api_test.go
> <https://github.com/ubuntu-core/snappy/pull/33#discussion_r43409099>:
>
> > @@ -134,8 +130,7 @@ oem: {store: {id: %q}}
> >  }
> >
> >  func (s *apiSuite) TestPackageInfoOneIntegration(c *check.C) {
> > -	d := New()
> > -	d.addRoutes()
> > +	newTestDaemon()
>
> The branch looks fine as it's not really making matters any worse, but the
> lack of context here indicates some disturbing use of global state. How
> come this creates something that is dropped on the floor, yet it somehow
> manages to get picked up by the logic below? And finalized somehow?
>
> I really hope we can get to a point where there's no global state, and the
> setup an termination are clear and in context.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ubuntu-core/snappy/pull/33/files#r43409099>.
>



-- 
John Lenton (jlenton@gmail.com) ::: http://chipaca.com